src/pj_fwd.c

```diff
@@ -174,11 +174,23 @@ static PJ_COORD fwd_finalize (PJ *P, PJ_COORD coo) {
 }
 
 
+static PJ_COORD error_or_coord(PJ *P, PJ_COORD coord, int last_errno) {
+    if (proj_errno(P))
+        coord = proj_coord_error();
+    else
```

Comment:
Could handle the non-common case by  `if (...) return proj_coord_error()` here.

This would raise the common case to top level, rather than partially depressed behind an else clause